aten/src/ATen/native/cpu/Loops.h

```diff
@@ -226,8 +230,9 @@ void cpu_kernel_vec(TensorIterator& iter, func_t&& op, vec_func_t&& vop) {
 template <typename func_t>
 void cpu_serial_kernel(TensorIterator& iter, func_t&& op, const Range& range) {
   using traits = function_traits<func_t>;
-  TORCH_INTERNAL_ASSERT((std::is_void<typename traits::result_type>::value &&
-    iter.noutputs() == 0 && iter.ntensors() == traits::arity) || (iter.ntensors() >= traits::arity + 1));
+  constexpr bool result_void = std::is_void<typename traits::result_type>::value;
+  TORCH_INTERNAL_ASSERT(iter.ninputs() == traits::arity &&
```

Comment:
This is a different condition than used to be here (previously, result of void type was valid with non-zero noutputs, now it is not). Is this intended?

Reply:
yes, I meant these to be as strict as possible (while still passing the tests).  I'm totally happy to loosen these later if we need to, but they should be based on what we actually need.